bundles/statistics/statsgrid2016/components/IndicatorDataForm.js

```diff
@@ -95,5 +117,50 @@ Oskari.clazz.define('Oskari.statistics.statsgrid.IndicatorDataForm', function (l
             }
         });
         return data;
+    },
+    openImportPopup: function () {
+        var me = this;
+        var popup = Oskari.clazz.create('Oskari.userinterface.component.Popup');
+        popup.makeDraggable();
+        var content = jQuery(this.__templates.import({
+            placeholder: me.locale('userIndicators.import.placeholder')
+        }));
+        var okBtn = Oskari.clazz.create('Oskari.userinterface.component.buttons.OkButton');
+
+        okBtn.setPrimary(true);
+        okBtn.setHandler(function () {
+            var textarea = content.find('textarea');
+            var data = me.parseUserData(textarea);
+            me.trigger('import.user.data', data);
```

Comment:
Since the import is part of the data form I'd just call the fillTable() from here. It's kinda unnecessary complexity to check where the event is being listened since it only affects this form. So remove the event and just call me.fillTable(data);